Feature/rlp-703/invoicing

mauricioirace opened the pull request:
### Mediated
![mediated](https://user-images.githubusercontent.com/9200315/90186492-9e59c800-dd8e-11ea-8726-1436d98529d7.gif)
### Invoice
![invoice](https://user-images.githubusercontent.com/9200315/90186508-a31e7c00-dd8e-11ea-8d3c-5b671c0f3850.gif)
### p2p
![p2p](https://user-images.githubusercontent.com/9200315/90186521-a6b20300-dd8e-11ea-8cc7-99cc79854359.gif)

### Quick Payment

![quick-payment](https://user-images.githubusercontent.com/9200315/90437629-14b93b80-e0a9-11ea-9ce2-68bebfeed510.gif)

Changes:
- Add invoicing to web ui

gasparmedina committed:
Merge pull request #1 from rsksmart/releases/0.0.2

Releases/0.0.2

Commit:
[ADD] support payments with invoices

Commit:
[FIX] url to put channels

Commit:
[FIX] remove rns changes

gasparmedina committed:
log are removed

gasparmedina committed:
log are removed

Commit:
[FIX] pull requests improvements

Commit:
Merge branch 'billing' of https://github.com/gasparmedina/lumino-web into billing

marcosmartinez7 committed:
Merge pull request #2 from gasparmedina/billing

Pay Invoice Support

Commit:
Add '()' to web3js functions

Commit:
Merge branch 'fix/RLP-746/tokens' of https://github.com/thelagger69/lumino-web into feature/RLP-703/invoicing

mauricioirace commented on src/actions/paymentsActions.js:
```diff
@@ -152,6 +152,32 @@ export const quickPayment = (partnerAddress, tokenAddress, totalDeposit) => asyn
+export const invoicePayment = (invoice) => async (dispatch,
+    getState) => {
+    await getTokenApp("invoicePayment");
+    new Promise((resolve, reject) =>
```
I think this Promise is unnecessary, and makes the functions a bit extra-verbose

marcosmartinez7 replied:
Yeah me too, we should improve this in the short time, because there are other functions that handle the async calls like this too

mauricioirace commented on src/components/billing/InvoiceForm.jsx:
```diff
@@ -0,0 +1,27 @@
+import React, {Component} from "react";
+
+export default class InvoiceForm extends Component {
```
Being this a stateless component, I think a functional component would be more suitable, but it's fine anyway

marcosmartinez7 replied:
Yeah, i like the idea of having all the components defined exactly the same way. But as a function is ok too.

mauricioirace commented on .gitignore:
```diff
@@ -22,3 +22,4 @@ chrome-user-data
 .vscode
 *.swp
 *.swo
+public/static/endpointConfig.js
```
Why was this file removed?

marcosmartinez7 replied:
This must not be ignored. It must be ignored only on the node, because its autogenerated on each run, but not in the web proyect.

marcosmartinez7 replied:
Also, it must not be removed

mauricioirace commented on src/containers/payments/QuickPayment.jsx:
```diff
@@ -106,63 +121,109 @@ class QuickPayment extends Component {
   };
 
+  handleQuickPaymentInvoice = () => {
+    const { invoicePayment } = { ...this.props };
```
Isn't this the same as: `const { invoicePayment } = this.props;` ?

marcosmartinez7 replied:
Yep, its redundant to clone the right side

marcosmartinez7 commented on src/restClient/index.js:
```diff
@@ -2,7 +2,7 @@ import axios from "axios";
 
 const client = axios.create({
-  baseURL: isDevelopment() ? "http://localhost:5001" : window.luminoUrl
+  baseURL: isDevelopment() ? "http://localhost:5002" : window.luminoUrl
```
This must not be modified, 5001 looks good as default port

marcosmartinez7 reviewed (changes requested):
Lgtm in general, just some minor changes.

Also, can you put a quick video on the PR description showing this three cases working?:

- Invoice payment
- p2p payment
- Mediated payment

Commit:
Add minor style fixes

- Removed endpointConfig from .gitignore
- Replaced unnecesary promise by async/await

marcosmartinez7 reviewed (changes requested):
Thanks for de videos. There are some things left:

1. This prs affects the QuickPayment component too, that component is on the header, near the notification buttom, did you test that? 
2. Can you put a video about that functionallity?
3. There are some LGTM bot warnings to fix

marcosmartinez7 commented:
test

Commit:
Remove unused consts and imports

mauricioirace commented:
> Thanks for de videos. There are some things left:
> 
> 1. This prs affects the QuickPayment component too, that component is on the header, near the notification buttom, did you test that?
> 2. Can you put a video about that functionallity?
> 3. There are some LGTM bot warnings to fix

I've just added a video showing QuickPayment functionality, as fixed LGTM warnings
